api/admin/controller.py

```diff
@@ -1045,20 +1045,24 @@ def _set_integration_settings_and_libraries(self, integration, protocol):
                     value = json.dumps(value)
             else:
                 value = flask.request.form.get(key)
-            if setting.get("options") and value not in [option.get("key") for option in setting.get("options")]:
-                self._db.rollback()
-                return INVALID_CONFIGURATION_OPTION.detailed(_(
-                    "The configuration value for %(setting)s is invalid.",
-                    setting=setting.get("label"),
-                ))
+            if value and setting.get("options"):
+                # This setting can only take on values that are in its
+                # list of options.
+                allowed = [option.get("key") for option in setting.get("options")]
+                if value not in allowed:
+                    self._db.rollback()
+                    return INVALID_CONFIGURATION_OPTION.detailed(_(
+                        "The configuration value for %(setting)s is invalid.",
+                        setting=setting.get("label"),
+                    ))
```

Comment:
I changed this so that an optional value with 'options' doesn't have to set a value.